packages/app-desktop/gui/MainScreen/commands/showSpellCheckerMenu.ts

```diff
@@ -30,8 +30,9 @@ export const runtime = (): CommandRuntime => {
 			const s: string[] = [];
 			// eslint-disable-next-line github/array-foreach -- Old code before rule was applied
 			languages.forEach((language: string) => {
-				s.push(language.split('-')[0]);
```

Comment:
Hi, I am sorry that I did not discuss my changes before. I saw in the docs that contributions should be first addressed and discussed in the GitHub issues or Joplin forum. In the future, I will be more aware of that and provide more explicit explanations.

Reply:
I added only `s.push(language)` and removed `s.push(language.split('-')[0]` because to explicitly display the spellcheck languages with different variants like  "en-AU, en-GB, en-US" instead of "en, en, en". This is something that [tomasz1986](https://github.com/tomasz1986) wanted to see and I agree that from the UI/UX perspective, it is better this way.

The changes should not affect `SpellCheckerService.ts` and the `state: AppState`.

I might have missed something based on your review, let me know!

Reply:
Here are a few other options:

## 1. Only show the region code when it prevents duplicate values from being added to `s`

Another option might be to only include the region code when there are multiple copies of the same language.

For example,
- `languages = [ 'en-GB', 'en-US' ]` => `s = [ 'en-GB', 'en-US' ]`
- `languages = [ 'en-GB', 'es-MX' ]` => `s = [ 'en', 'es' ]`.

## 2. Simplify `[ 'en', 'en', 'en' ]` to `[ 'en' ]`

This is based on one of the [list items in "describe what you expected to happen" in the original issue](https://github.com/laurent22/joplin/issues/7834).

Reply:
> ## 2. Remove duplicates
> For example,
> 
> * `languages = [ 'en', 'en', 'en' ]` => `s = [ 'en' ]`
> 
> This is based on one of the [list items in "describe what you expected to happen" in the original issue](https://github.com/laurent22/joplin/issues/7834).

As the author of the original issue, I actually like this method a lot. It would allow for the language list to take less space, so that more title text could be displayed on the screen, especially on narrower displays.

Reply:
I agree, the second solution is good in terms of usability and reduces clutter. @HahaBill, this is what I was trying to get at: first find out why something was done in a particular way. It was indeed to reduce clutter because putting the language as eg fr-FR or it-IT doesn't help the user.

So the idea would be to keep the clutter to a minimum while solving the original bug report. Actually both solution 1 or 2 would work but indeed 2 seems better.

Reply:
For me, the second solution also looks better than the first one, and everyone seems to lean towards that. The second solution it is! :)

(at sign)laurent22 I see now! That is a really good point and lesson learned. I will keep that in mind.

Reply:
Here's the solution:

<img width="694" alt="Screenshot 2024-02-23 at 22 41 55" src="https://github.com/laurent22/joplin/assets/32552798/34699f94-d732-4881-896c-2552a4bc8be7">